app/src/main/kotlin/no/nav/k9punsj/kafka/KafkaHendelseProducer.kt

```diff
@@ -6,47 +6,31 @@ import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.kafka.core.KafkaTemplate
-import org.springframework.kafka.support.SendResult
 import org.springframework.stereotype.Component
-import org.springframework.util.concurrent.ListenableFuture
-import org.springframework.util.concurrent.ListenableFutureCallback
 
 @Component
 @IkkeTestProfil
 class KafkaHendelseProducer(
     @Qualifier(AIVEN) private val kafkaTemplate: KafkaTemplate<String, String>
 ) : HendelseProducer {
     override fun send(topicName: String, data: String, key: String) {
-        val future: ListenableFuture<SendResult<String?, String?>> = kafkaTemplate.send(topicName, key, data)
-        future.addCallback(object : ListenableFutureCallback<SendResult<String?, String?>?> {
-            override fun onSuccess(result: SendResult<String?, String?>?) {
-                val recordmetadata = result?.recordMetadata
-                logger.info("Melding sendt OK på Topic=[$topicName], Key=[$key], Offset=[${recordmetadata?.offset()}, Partition=[${recordmetadata?.partition()}]")
+        kafkaTemplate.send(topicName, key, data).also {
+            when(it.isDone) {
```

Comment:
Litt usikker på om dette blir riktig.

Reply:
KafkaTemplate.send() returnerer en CompletableFuture.
Du kan dermed bruke noe liknende for å håndtere callback:

`kafkaTemplate.send(topicName, key, data)
            .exceptionally { ex: Throwable ->
                logger.warn...
                throw ex
            }
            .thenAccept { result: SendResult<String, String> ->
                log.info(
                    "Sendte melding med offset {} på {}",
                    result.recordMetadata.offset(),
                    result.producerRecord.topic()
                )
            }`

Reply:
NP :)